src/reader.cr

```diff
@@ -1,23 +1,35 @@
-def remove_comments(sql)
-  sql = sql.lines[1..-1].join "\n"
-  sql.gsub(/^\s*--.*?\n/, "").strip
+def isMeta(line)
+  line.match(/^\s*-- name: ([a-z\_\?\!]+?\(.*?\)).*?\n/)
 end
 
-def get_metadata(sql)
-  sql.lines[0].gsub(/^\s*-- name: ([a-z\_\?\!]+?\(.*?\)).*?\n/) do |token, match|
+def isComment(line)
+  line.match(/^\s*--.*/)
+end
+
+def parse(lines)
+  lines.reject {|x| isComment(x) && !isMeta(x) || x == "\n"}
```

Comment:
Also you should use `line` variable instead of `x` for readability issues.

```
lines.reject {|line| comment?(line) && !meta?(line) || line == "\n"}
```